src/app/shell/header/language-switch/language-switch.component.html

```diff
@@ -17,7 +17,7 @@
         <ul *ngIf="availableLocales.length">
           <ng-container *ngFor="let l of availableLocales">
             <li *ngIf="l.lang !== locale.lang">
-              <a [href]="location | makeHref: { redirect: '1', lang: l.lang }"> {{ l.displayName }} </a>
+              <a [href]="location | makeHref: { redirect: '1', lang: l.lang } | async"> {{ l.displayName }} </a>
```

Comment:
I guess the `redirect` functionality is not really needed here.
Even if the previous behavior with an `undefined` `multiSiteLocaleMap` is used will the language parameter vanish as soon as the user routes after the language switch.